src/iuse.cpp

```diff
@@ -7781,3 +7781,48 @@ int iuse::magnesium_tablet( player *p, item *it, bool, const tripoint & )
     p->add_effect( effect_magnesium_supplements, 16_hours );
     return it->type->charges_to_use();
 }
+
+int iuse::coin_flip( player *p, item *it, bool, const tripoint & )
```

Comment:
You're right, there's no reason for them to have a return type of int. I don't know about returning void, but I do know about not returning. Lots of existing iuse functions have a return type of int for no particular reason, not that it hurts any other part of the code.

Reply:
This would involve writing another overload for `add_iuse` in `item_factory.h`, as the current one only accepts `use_function_pointer`, which is defined as:
```
typedef int ( iuse::*use_function_pointer )( player *, item *, bool, const tripoint & );
```

Probably some other architectural changes required I don't know about. I'm not too familiar with the code beyond that.

Reply:
@Night-Pryanik

> Is there a sense in returning int for these new functions when we don't use returned value at all? Isn't returning void better?

It's strange you complain about the apparently unused return value, but you do not complain about the unused parameters.

The function has to follow a specific interface definition which requires exactly those parameters and that return type (C++ is a statically typed language).

Fixing this would require actions indicated by Muffindrake, which would also make the whole interface much more complicated with little to no gain.